source/common/upstream/health_checker_impl.cc

```diff
@@ -134,6 +134,7 @@ void HttpHealthCheckerImpl::HttpActiveHealthCheckSession::onEvent(Network::Conne
   }
 }
 
+// TODO(lilika) : Support connection pooling
```

Comment:
@mattklein123 are we correct in reading this code and thinking there is no connection pooling for HTTP health checks?

Reply:
Correct. HTTP HCs are very simple. LMK if you want me to review this change (I didn't look at it).

Reply:
No worries; I think this is fine to land as is, it's more to better understand how good/bad it is to keep nuking the HdsClusters on each update; seems fine given we don't have any stateful LB or concerns around connection pooling today.